src/App.tsx

```diff
@@ -7,9 +7,12 @@ import Footer from './components/Footer/Footer';
 import { filterPostsByGenre, filterPostsByName } from './utils/filterPosts';
 import sortPosts from './utils/sortPost';
 import { SortingValue } from './types/types';
+import Search from './components/Search/Search';
+import Modal from './components/Modal/Modal';
+import { AppProvider } from './Context';
 
 import './App.scss';
-import Search from './components/Search/Search';
+import ModalManager from './ModalManager/ModalManager';
 
 const App = () => {
   const [searchQuery, setSearchQuery] = useState('');
```

Comment:
1) Let's create Content component and move all components and logic there.
2) Let's use App for all providers, routers, etc and rendering just one Content component